src/lib/uptime_service/uptime_service.ml

```diff
@@ -75,60 +75,101 @@ let send_uptime_data ~logger ~interruptor ~(submitter_keypair : Keypair.t) ~url
+                        |> Cohttp_async.Body.of_string )
+                      url)
+               with
+               | {status; _}, body ->
+                   let succeeded = Cohttp.Code.code_of_status status = 200 in
+                   ( if succeeded then
+                     [%log info]
+                       "Sent block with state hash $state_hash to uptime \
+                        service at URL $url"
+                       ~metadata:
+                         [ ("state_hash", State_hash.to_yojson state_hash)
+                         ; ( "includes_snark_work"
+                           , `Bool (Option.is_some block_data.snark_work) )
+                         ; ("is_produced_block", `Bool produced)
+                         ; ("url", `String (Uri.to_string url)) ]
+                   else if attempt >= max_attempts then
+                     let base_metadata =
+                       [ ("state_hash", State_hash.to_yojson state_hash)
+                       ; ("url", `String (Uri.to_string url))
+                       ; ("http_code", `Int (Cohttp.Code.code_of_status status))
+                       ; ( "http_error"
+                         , `String (Cohttp.Code.string_of_status status) )
+                       ; ("payload", json) ]
+                     in
+                     let extra_metadata = metadata_of_body body in
+                     let metadata = base_metadata @ extra_metadata in
+                     [%log error]
+                       "After %d attempts, failed to send block with state \
+                        hash $state_hash to uptime service at URL $url, no \
+                        more retries"
+                       max_attempts ~metadata
+                   else
+                     let base_metadata =
                        [ ("state_hash", State_hash.to_yojson state_hash)
-                       ; ( "includes_snark_work"
-                         , `Bool (Option.is_some block_data.snark_work) )
-                       ; ("is_produced_block", `Bool produced)
-                       ; ("url", `String (Uri.to_string url)) ]
-                 else
-                   let base_metadata =
+                       ; ("url", `String (Uri.to_string url))
+                       ; ("http_code", `Int (Cohttp.Code.code_of_status status))
+                       ; ( "http_error"
+                         , `String (Cohttp.Code.string_of_status status) ) ]
+                     in
+                     let extra_metadata = metadata_of_body body in
+                     let metadata = base_metadata @ extra_metadata in
+                     [%log info]
+                       "Failure when sending block with state hash \
+                        $state_hash to uptime service at URL $url, attempt %d \
+                        of %d, retrying"
+                       attempt max_attempts ~metadata ) ;
+                   succeeded
+             in
+             match%map.Deferred.Let_syntax
```

Comment:
This interruptible should ideally be linked with the outer interruptible, otherwise it's serving no purpose here.

Reply:
Simpler than what I proposed over slack: use `Interruptible.uninterruptible` inside the monitor to get a `_ Interruptible.t Deferred.t`, `uninterruptible` to convert it, and `Interruptible.join` outside of the monitor.